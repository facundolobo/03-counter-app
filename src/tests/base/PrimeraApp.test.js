import { getByText, render } from '@testing-library/react';
import { shallow } from 'enzyme';
import React from 'react';
import '@testing-library/jest-dom'


const { default: PrimeraApp } = require("../../PrimeraApp")


describe('Pruevas en <PrimeroApp>', () => {
    // test('debe mostrar el mensaje "Hola, soy Goku"',()=> {
        

    //     const saludo ="Hola, soy Goku"
    //     const {getByText} = render (<PrimeraApp saludo= {saludo}/>)
        
    //     expect (getByText (saludo)).toBeInTheDocument();


    //     })
    test('debe de mostrar <PrimeraApp /> correctamente', () => {
        const saludo = 'Hola, soy Goku';
        const wrapper = shallow (<PrimeraApp saludo={saludo} />);
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe mostrar el subtitulo enviado por props', () => {
        
        const saludo = 'Hola, soy Goku';
        const subTitulo= 'Soy un subtitulo';
        
        const wrapper = shallow (
        <PrimeraApp 
        saludo={saludo} 
        subtitulo={subTitulo}
        />
        );
        
        const textoParrafo = wrapper.find('p').text();
        expect(textoParrafo).toBe(subTitulo);

    })
    
    
    }) 

