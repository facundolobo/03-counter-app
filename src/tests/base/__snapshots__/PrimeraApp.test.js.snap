// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruevas en <PrimeroApp> debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, soy Goku
    !!!
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
